lib/internal/url.js

```diff
@@ -1392,6 +1392,20 @@ function getPathFromURLPosix(url) {
   return decodeURIComponent(pathname);
 }
 
+function getPathFromURLString(path) {
+  if (typeof path === 'string') {
+    if (!path.startsWith('file://'))
```

Comment:
A possible micro-optimization here would be to make this `path.length < 7 || !path.startsWith('file://')`.

Also, is it intentional to be enforcing a lowercase protocol like that?